tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -121,12 +121,10 @@ private static void LogMatchesIfDebugEnabled(IReadOnlyCollection<object>? result
 
             Log.Error(ex, "Call into the security module failed with arguments {Args}", stringBuilder.ToString());
         }
-        finally
+
+        if (_localRootSpan.Context.TraceContext is not null)
         {
-            // annotate span
-            _localRootSpan.SetMetric(Metrics.AppSecEnabled, 1.0);
-            _localRootSpan.SetTag(Tags.AppSecRuleFileVersion, _security.WafRuleFileVersion);
-            _localRootSpan.SetTag(Tags.RuntimeFamily, TracerConstants.Language);
+            _localRootSpan.Context.TraceContext.WafExecuted = true;
```

Comment:
do we need this new property? Couldn't we simply add these tags if appsec is enabled in any case? (as, if appsec is enabled, there will be at least one waf run necessarily) I think it was added here for convenience reason but we don't neccesarily need to afaik